Homepage and Events Adjustments

GMSoudersJr opened the pull request:
# Homepage

### PROBLEM:

![fCCSeoul live homepage](https://github.com/Free-Code-Camp-Seoul/fcc-seoul-website/assets/44438849/dd860af8-73a6-40be-a603-fa1b0043dc4c)
- The page container seems to be pushed down because the lottie animation overflows its flexbox.

### FIX:


![fCCSeoul local homepage update](https://github.com/Free-Code-Camp-Seoul/fcc-seoul-website/assets/44438849/fe9ab958-d6d1-4b7e-9d25-2708ddaad8c4)
- Made CSS variables and used useRef to compute actual size of viewport, navbar, footer, title, and meetup info. Then resized the lottie animation container to fit in the flexbox . This was my React workaround. There must be a simpler way! The animation is smaller, not sure if that is desirable, though.

# Events Page

### Problem:

![fccSeoul local events page main branch](https://github.com/Free-Code-Camp-Seoul/fcc-seoul-website/assets/44438849/601e2c49-ffff-411a-b02a-5d7923222142)
- The page is larger than the window so the footer is not visible

### Fix:

![fccSeoul local events page updated branch](https://github.com/Free-Code-Camp-Seoul/fcc-seoul-website/assets/44438849/c39ed88f-750e-4554-aa63-df165062414f)
- Changed font sizes and weights since Poppins font is larger than the default font. Attempted to keep the sizing in the same style as the original and fit everything on one page.


# Importing CSS Issue
![fCC Seoul font issue](https://github.com/Free-Code-Camp-Seoul/fcc-seoul-website/assets/44438849/abf65810-1f73-4816-9da5-bbc60bba5f62)
- This is an issue with importing the font. This caused the difference between the fonts in my dev server and the production server. I do not know how to fix this at the moment and wonder if it will still be an issue upon merging. 🤷‍♂️

GMSoudersJr committed:
fix(homepage layout): fit animation in flexbox

Added css variables and implemented useRef to dynamically set the
height for lottie animation and homepage containers based on
viewport height.

GMSoudersJr committed:
fix(events page): fit to single page

Adjust typograpghy to fit on a single page since Poppins font-family is
larger than default. Tried to keep with the sizing and weight of
original page.

GMSoudersJr commented:
The Members page is longer than one page. I figured it does not need much work since it will hopefully have more members than one page can hold. 😃

ivanms1 commented on src/pages/Home/Home.module.scss:
could you try this instead, I think it's a simpler solution
```suggestion
    height: 45vh;
    margin-bottom: -50%;
```

and below in the media query

```css
 & .AnimationContainer {
      height: 30vh;
    }
```

you could also reduce the navbar and footer padding to `1rem` to get a bit more room.

ivanms1 commented on src/pages/Home/Home.tsx:
This might not be needed with the css added above

ivanms1 reviewed (commented):
Great job 🚀 
left some comments.

GMSoudersJr commented on src/pages/Home/Home.module.scss:
Definitely simpler. I was unsure how much I should change. Thanks for the feedback

ivanms1 commented:
thanks for noticing the font issue, I totally missed that one.
Perhaps we could load the font [this way instead](https://nextjs.org/docs/pages/building-your-application/optimizing/fonts)
but should be in a different PR.

ivanms1 commented on src/pages/Home/Home.module.scss:
if you want all the way you could

in Navbar and Footer
- reduce the padding to `1rem`
- ~fix the height to `56px` in both~ actually this is not needed

in `Home.module.scss`
- remove `justtify-content: center` from the `.Container` class

and finally in `Layout.module.scss`
```css
.Layout {
  align-items: center;
  background: linear-gradient(135deg, #211cab, #c74a95);
  background-size: cover;
  display: flex;
  flex-direction: column;
  /* add this line */
  height: 100vh;

  & .Content {
    flex: 1;
    max-width: 1344px;
    /* and this line */
    overflow: auto;
    padding: 3rem;
    width: 100%;
  }
}
```
this make only the content scrollable, leaving the navbar and footer always visible.

GMSoudersJr committed:
fix(layouts): home and events pages

Remove the logic from home.tsx. Revert to using CSS instead. Adjust
layout so navbar and footer are always visible. Shrink events
description to max-width of 600px for UX on wide screens.

GMSoudersJr commented:
New Commits follow your suggestions.

### Navbar and Footer
- Reduced padding

### Layout
- Navbar and Footer are always visible. Scroll through content only.

### Home
![fccSeoul local update 2](https://github.com/Free-Code-Camp-Seoul/fcc-seoul-website/assets/44438849/779c3474-9197-4030-8bfa-ed2ac087f61a)
- Looking sleeker!
- Animation is larger as well! Good call!

### Events
![fCCSeoul Events local update 2](https://github.com/Free-Code-Camp-Seoul/fcc-seoul-website/assets/44438849/df677664-97b1-4268-a1dd-0f65226a9ba6)
- Set `.description { max-width: 600px }` to follow a UX best practice of not having long lines of text.
- Reverted back to previous font sizes because the page will be longer than 100vh.

ivanms1 reviewed (approved):
LGTM 🚀
